### ISSUE TITLE
bug: cwd doesn't resolve correctly for sourceFile's relative to the call site

### ISSUE THREAD

**Issue by dmeehan1968** [Stedi/ts2asl#890]
When specifying `sourceFile`, there is a default assumption that its an absolute or relative path to the `process.cwd`.  This can be overridden by also specifying the `cwd` property (which could possibly be done with `__dirname`).

However, setting `cwd` to `__dirname` does not appear to work, as the cdk synth process emits:

```
Error: Cannot find entry file at {sourceFile as specified}
```

However, using `__dirname` along with `path.resolve` to create an absolute path does work:

```
sourceFile: path.resolve(__dirname, '{relative path to source}`)
```

Basically, use of `cwd` doesn't seem to produce the desired result when specifying a path relative to the call site.
